tests/std/tests/P3349R1_contiguous_iterators_to_pointers/test.cpp

```diff
@@ -0,0 +1,463 @@
+        if (current_ptr < range_first_ptr || current_ptr >= range_last_ptr) {
+            if constexpr (nothrow) {
+                safe_iter_nothrow_OOB_sign = true;
+            } else {
+                throw safe_iter_out_of_bounds_err{};
+            }
+        }
+        return *current_ptr;
+    }
+    constexpr pointer operator->() const noexcept {
+        if constexpr (nothrow) {
+            if (current_ptr < range_first_ptr || current_ptr > range_last_ptr) {
+                safe_iter_nothrow_OOB_sign = true;
+            }
+        }
+        return current_ptr;
+    }
+    constexpr reference operator[](difference_type n) const noexcept(nothrow) {
+        return *(*this + n);
+    }
+
+    constexpr safe_iter& operator+=(difference_type offset) noexcept(nothrow) {
+        current_ptr += offset;
+        if constexpr (!nothrow) {
+            if (current_ptr < range_first_ptr || current_ptr > range_last_ptr) {
+                throw safe_iter_out_of_bounds_err{};
+            }
+        }
+        return *this;
+    }
+    constexpr safe_iter operator+(difference_type offset) const noexcept(nothrow) {
+        auto tmp = *this;
+        tmp += offset;
+        return tmp;
+    }
+    friend constexpr safe_iter operator+(difference_type offset, const safe_iter& right) noexcept(nothrow) {
+        return right + offset;
+    }
+    constexpr safe_iter& operator-=(difference_type offset) noexcept(nothrow) {
+        return *this += -offset;
+    }
+    constexpr safe_iter operator-(difference_type offset) const noexcept(nothrow) {
+        return *this + (-offset);
+    }
+    constexpr difference_type operator-(const safe_iter& right) const noexcept {
+        return current_ptr - right.current_ptr;
+    }
+
+    constexpr safe_iter& operator++() noexcept(nothrow) {
+        return *this += 1;
+    }
+    constexpr safe_iter& operator--() noexcept(nothrow) {
+        return *this -= 1;
+    }
+    constexpr safe_iter operator++(int) noexcept(nothrow) {
+        auto temp = *this;
+        ++*this;
+        return temp;
+    }
+    constexpr safe_iter operator--(int) noexcept(nothrow) {
+        auto temp = *this;
+        --*this;
+        return temp;
+    }
+
+    constexpr bool operator==(const safe_iter& right) const noexcept {
+        return current_ptr == right.current_ptr;
+    }
+    constexpr auto operator<=>(const safe_iter& right) const noexcept {
+        return current_ptr <=> right.current_ptr;
+    }
+
+    constexpr difference_type operator-(pointer right) const noexcept {
+        return current_ptr - right;
+    }
+    friend constexpr difference_type operator-(pointer left, const safe_iter& right) noexcept {
+        return left - right.current_ptr;
+    }
+    constexpr bool operator==(pointer right) const noexcept {
+        return current_ptr == right;
+    }
+    friend constexpr bool operator==(pointer left, const safe_iter& right) noexcept {
+        return left == right.current_ptr;
+    }

```

Comment:
`operator==(pointer left, const safe_iter& right)` shouldn't be necessary, as C++20 equality rewriting will call `operator==(pointer right)`.